File: Data Structures & Algorithms/two-integer-sum-ii/submission-8.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {

        //do

        let L = 0, R = numbers.length - 1;

        while(L < R){
            const [left, right] = [numbers[L], numbers[R]];

            if(left + right === target) return [L + 1, R + 1];

            if(left + right < target) L++;

            if(left + right > target) R--;
        }

        return [];
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-8.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()

        res = []
        seen = set()
        for i, target in enumerate(nums):
            
            left = i + 1
            right = len(nums) - 1
   



            while left < len(nums) and left != right:
                sum = nums[left] + nums[right]
                lst = [target, nums[left], nums[right]]
                hsh = tuple(lst)
                if sum == -target and hsh not in seen :
                    res.append(lst)
                    seen.add(hsh)
                    left += 1
                elif sum > -target:
                    right -= 1
                else:
                    left += 1
        return res


            